tools/version-packages.sh

```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
```

Comment:
```suggestion
#!/usr/bin/env bash

# This script requires Bash 4+, since we want a version of Bash that supports mapfile.
if [ -z "${BASH_VERSINFO}" ] || [ -z "${BASH_VERSINFO[0]}" ] || [ ${BASH_VERSINFO[0]} -lt 4 ]; then
	echo "This script requires Bash version >= 4."
	read -p "Do you want to install it on your system with Homebrew (macOS)? [y/N]" -n 1 -r
	if [[ $REPLY != "y" && $REPLY != "Y" ]]; then
		exit 1;
	else
		brew install bash 2>/dev/null
		echo "Done!"
	fi
fi
```

Reply:
P.S. Yes, the change I made is applying the requirement for Bash 4+ to all the tools I'm touching here, even the ones that don't need it. That's intentional.